spinegeneric/cli/generate_figure.py

```diff
@@ -467,6 +470,19 @@ def main():
     args = get_parameters()
     display_individual_subjects = args.indiv_subj
 
+    # check if input yml file exists
+    if os.path.isfile(args.config):
+        fname_yml = args.config
+    else:
+        sys.exit("ERROR: Input yml file {} does not exist or path is wrong.".format(args.config))
+
+    # fetch input yml file as dict
+    with open(fname_yml, 'r') as stream:
+        try:
+            dict_exclude_subj = yaml.safe_load(stream)
+        except yaml.YAMLError as exc:
+            print(exc)
+
```

Comment:
Maybe, we could add condition for initialization of empty `dict_exclude_subj` dict, when no input yml config file is passed (I know, that in most cases some subjects will be excluded and yml file will be passed, but for sure, it is better to deal with situation when no subject will be excluded):

```suggestion
    # create dict with subjects to exclude if input yml config file is passed
    if args.config is not None:
        # check if input yml file exists
        if os.path.isfile(args.config):
            fname_yml = args.config
        else:
            sys.exit("ERROR: Input yml file {} does not exist or path is wrong.".format(args.config))

        # fetch input yml file as dict
        with open(fname_yml, 'r') as stream:
            try:
                dict_exclude_subj = yaml.safe_load(stream)
            except yaml.YAMLError as exc:
                print(exc)
    else:
        # initialize empty dict if no config yml file is passed
        dict_exclude_subj = dict()

```

Reply:
great suggestion @valosekj . i need to resolve the conflicts first, then i'll commit your suggestion